pyodide_build/mkpkg.py

```diff
@@ -134,11 +185,17 @@ def make_parser(parser):
 
 
 def main(args):
-    package = args.package[0]
-    if args.update:
-        update_package(package)
-        return
-    make_package(package, args.version)
+    try:
+        package = args.package[0]
+        if args.update:
+            update_package(package, update_patched=True)
+            return
+        if args.update_if_not_patched:
+            update_package(package, update_patched=False)
+            return
+        make_package(package, args.version)
+    except MkpkgFailedException as e:
```

Comment:
Not sure about this. If something fails, I would rather get the actual exception rather than a simple error message saying that it failed.

Reply:
I only caught two types of errors this way:
1. When the request fails, it produces messages like:
Failed to load metadata for libxslt from https://pypi.org/pypi/libxslt/json: HTTP Error 404: Not Found

2. If there is no sdist it prints an error message like:
No sdist URL found for package swiglpk (https://pypi.org/project/swiglpk/)

All unexpected exceptions are not a `MkpkgFailedException` and so raise a different exception. I just want to avoid cluttering the output with tracebacks from common failures.

Reply:
OK, fair enough. Thanks for the additional information.

Reply:
I'll add this as a comment.